engine/evarutil.ml

```diff
@@ -499,12 +499,12 @@ let e_new_type_evar env evdref ?src ?filter ?naming ?principal ?hypnaming rigid
     evdref := evd;
     c
 
-let new_Type ?(rigid=Evd.univ_flexible) env evd = 
+let new_Type ?(rigid=Evd.univ_flexible) evd =
   let open EConstr in
```

Comment:
Even though they're not used, I'd probably leave the environment arguments in engine for uniformity. Furthermore it might turn out necessary to have them at some point in the future, so it doesn't hurt to leave them.

Reply:
>Even though they're not used, I'd probably leave the environment arguments in engine for uniformity. Furthermore it might turn out necessary to have them at some point in the future, so it doesn't hurt to leave them.

I think it's more uniform to avoid unused arguments. Why do you think they may become used in the future?

Reply:
>global references

What does that have to do with new_Type and company?

Reply:
That was a specific example, but I'm just saying that in general it's unclear what the future API may require, and there have been quite a few such deep changes in the recent past for me to be cautious (like universe polymorphism and the like).

Reply:
I've always wondered why `env + sigma` were not packed together BTW.